theano/compile/ops.py

```diff
@@ -346,6 +346,18 @@ def __init__(self, i):
         i = int(i)
         self.i = i
 
+    # NB:
+    # 1) params_type is defined as a property to avoid
+    #    loop in Python import caused by importing theano.scalar below
+    #    when params_type is defined directly in class code.
+    # 2) We wrap scalar into ParamsType (instead of directly using scalar as op param)
+    #    to avoid Theano converting scalar param to constant that would be later
+    #    hardcoded as litteral in C code, making us loose all the advantages of
+    #    using params.
```

Comment:
Why it get converted to constant? Maybe that should be fixed. Not mandatory for this PR.